owsi-core/owsi-core-components/owsi-core-component-wicket-more/src/main/java/fr/openwide/core/wicket/more/export/file/component/FileDownloadAjaxLink.java

```diff
@@ -46,36 +50,21 @@ protected void updateAjaxAttributes(AjaxRequestAttributes attributes) {
 
 	@Override
 	public void onClick(AjaxRequestTarget target) {
-		File tmp = null;
 		try {
-			tmp = generateFile();
-			
-			if (tmp == null) {
-				throw new IllegalStateException("File is missing.");
-			}
-			
-			tempFileModel.setObject(tmp);
-			
+			tempFileInformationModel.setObject(fileInformationModel.getObject());
 			ajaxDownload.initiate(target);
-			
 			target.appendJavaScript(loadingPopup.closeStatement().render());
 		} catch (Exception e) {
-			LOGGER.error("Error while generating a file.", e);
-			
-			FileUtils.deleteQuietly(tmp);
-			tempFileModel.setObject(null);
-			
+			LOGGER.error("Error while downloading a file.", e);
```

Comment:
On second thought, you were right: it might not be a good idea to delete the file. In the case of temporary files, other cleaning mechanisms (like periodic cleaning) are required anyway (since the HTTP client might not follow the redirection), and not deleting the file allows us to point directly to stored files that should not deleted.